src/HttpHeader.cc

```diff
@@ -759,6 +773,16 @@ HttpHeader::addEntry(HttpHeaderEntry * e)
 
     debugs(55, 7, this << " adding entry: " << e->id << " at " << entries.size());
 
+    // verify whether adding the entry will not exceed the maximum header size allowed
+    const auto max = maxLen();
+    if (max != UnrestrictedSize) {
+        const auto eLen = e->length();
+        if (len + eLen > max) {
```

Comment:
Adjust to avoid overflows. They are very unlikely, but let's not create a concern.

```suggestion
        if (eLen > maxLen || len > maxLen - eLen) { // i.e. `len + eLen > maxLen` without overflows
```

Reply:
I removed this block after addressing another change [request](https://measurementfactory.atlassian.net/browse/SQUID-479?focusedCommentId=43555&oldIssueView=true&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-43555).